kolibri/core/assets/src/views/__tests__/FocusTrap.spec.js

```diff
@@ -0,0 +1,82 @@
+import { render, screen, fireEvent } from '@testing-library/vue';
+import userEvent from '@testing-library/user-event';
+import FocusTrap from '../FocusTrap.vue';
+import FocusTrapWrapper from './FocusTrapWrapper.vue';
+
+const renderComponent = (props = {}) => {
+  return render(FocusTrap, {
+    props: {
+      disabled: false,
+      ...props,
+    },
+  });
+};
+
+describe('FocusTrap', () => {
+  it('should emit the "shouldFocusFirstEl" element when the tab key is pressed once', async () => {
+    const { emitted } = renderComponent();
+
+    await userEvent.tab();
+    expect(emitted()).toHaveProperty('shouldFocusFirstEl');
+    expect(emitted().shouldFocusFirstEl.length).toBe(1);
+  });
+
+  it("should trap the focus ('shouldFocusFirstEl' should be emitted twice) on pressing tab twice ", async () => {
+    const { emitted } = renderComponent();
+
+    await userEvent.tab();
+    await userEvent.tab();
+
+    expect(emitted()).toHaveProperty('shouldFocusFirstEl');
+    expect(emitted().shouldFocusFirstEl.length).toBe(2);
+  });
+
+  it('should emit "shouldFocusLastEl" when the element is subsequently focused after the inital focus', async () => {
```

Comment:
I would change this to something like "should trap the focus and emit "shouldFocusLastEl" when the first element is focused and we focus the previous element"